async-nats/src/client.rs

```diff
@@ -43,6 +44,62 @@ impl From<tokio::sync::mpsc::error::SendError<Command>> for PublishError {
     }
 }
 
+pub struct Publish {
+    sender: mpsc::Sender<Command>,
+    subject: String,
+    payload: Bytes,
+    headers: Option<HeaderMap>,
+    respond: Option<String>,
+}
+
+impl Publish {
+    pub(crate) fn new(sender: mpsc::Sender<Command>, subject: String, payload: Bytes) -> Publish {
+        Publish {
+            sender,
+            subject,
+            payload,
+            headers: None,
+            respond: None,
+        }
+    }
+
+    pub fn headers(mut self, headers: HeaderMap) -> Publish {
+        self.headers = Some(headers);
+        self
+    }
+
+    pub fn reply(mut self, subject: String) -> Publish {
+        self.respond = Some(subject);
+        self
+    }
+}
+
+impl IntoFuture for Publish {
+    type Output = Result<(), PublishError>;
+    type IntoFuture = Pin<Box<dyn Future<Output = Result<(), PublishError>> + Send>>;
+
+    fn into_future(self) -> Self::IntoFuture {
+        let sender = self.sender.clone();
```

Comment:
Is this clone needed?

Reply:
Hey @paolobarbolini, thank you so much for the review. Much appreciated. Also good catch!